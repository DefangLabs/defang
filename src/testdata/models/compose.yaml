services:
  app:
    image: app
    models: # per docs, should support list as well
      ai_model: {}
  withendpoint:
    image: app
    models:
      my_model:
        endpoint_var: MODEL_URL

models:
  ai_model:
    model: ai/model
  my_model:
    model: ai/model
    context_size: 1024
    runtime_flags:
      - "--a-flag"
      - "--another-flag=42"
