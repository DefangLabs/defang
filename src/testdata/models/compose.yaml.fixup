{
  "ai_model": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "OPENAI_API_KEY": ""
    },
    "image": "defangio/openai-access-gateway",
    "networks": {
      "model_provider_private": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 80,
        "protocol": "tcp"
      }
    ]
  },
  "app": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "AI_MODEL_MODEL": "ai/model",
      "AI_MODEL_URL": "http://ai-model/api/v1/"
    },
    "image": "app",
    "models": {
      "ai_model": {}
    },
    "networks": {
      "default": null,
      "model_provider_private": null
    }
  },
  "my_model": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "OPENAI_API_KEY": ""
    },
    "image": "defangio/openai-access-gateway",
    "networks": {
      "model_provider_private": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 80,
        "protocol": "tcp"
      }
    ]
  },
  "withendpoint": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "MODEL_URL": "http://my-model/api/v1/",
      "MY_MODEL_MODEL": "ai/model"
    },
    "image": "app",
    "models": {
      "my_model": {
        "endpoint_var": "MODEL_URL"
      }
    },
    "networks": {
      "default": null,
      "model_provider_private": null
    }
  }
}