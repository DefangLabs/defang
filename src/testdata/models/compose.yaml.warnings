 ! service "ai_model": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "app": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "my_model": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "withendpoint": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
