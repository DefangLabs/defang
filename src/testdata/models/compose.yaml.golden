name: models
services:
  app:
    image: app
    models:
      ai_model: {}
    networks:
      default: null
  withendpoint:
    image: app
    models:
      my_model:
        endpoint_var: MODEL_URL
    networks:
      default: null
networks:
  default:
    name: models_default
models:
  ai_model:
    model: ai/model
  my_model:
    model: ai/model
    context_size: 1024
    runtime_flags:
      - --a-flag
      - --another-flag=42
